internal/core/src/common/Geometry.h

```diff
@@ -0,0 +1,176 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License
+#pragma once
+
+#include "ogr_geometry.h"
+#include <memory>
+#include <string_view>
+#include "common/EasyAssert.h"
+
+namespace milvus {
+
+class Geometry {
+ public:
+    Geometry() = default;
+
+    // all ctr assume that wkb data is valid
+    explicit Geometry(const void* wkb, size_t size) {
+        OGRGeometry* geometry = nullptr;
+        OGRGeometryFactory::createFromWkb(wkb, nullptr, &geometry, size);
+
+        AssertInfo(geometry != nullptr,
+                   "failed to construct geometry from wkb data");
+        geometry_.reset(geometry);
+        //to_wkb_internal();
+    }
+
+    explicit Geometry(const char* wkt) {
+        OGRGeometry* geometry = nullptr;
+        OGRGeometryFactory::createFromWkt(wkt, nullptr, &geometry);
+        AssertInfo(geometry != nullptr,
+                   "failed to construct geometry from wkt data");
+        geometry_.reset(geometry);
+        //to_wkb_internal();
+    }
+
+    Geometry(const Geometry& other) {
+        if (other.IsValid()) {
+            this->geometry_.reset(other.geometry_->clone());
+            //this->to_wkb_internal();
+        }
+    }
+
+    Geometry(Geometry&& other) noexcept
+        :  //wkb_data_(std::move(other.wkb_data_)),
+          //size_(other.size_),
+          geometry_(std::move(other.geometry_)) {
+    }
+
+    Geometry&
+    operator=(const Geometry& other) {
+        if (this != &other && other.IsValid()) {
+            this->geometry_.reset(other.geometry_->clone());
+            //this->to_wkb_internal();
+        }
+        return *this;
+    }
+
+    Geometry&
+    operator=(Geometry&& other) noexcept {
+        if (this != &other) {
+            //wkb_data_ = std::move(other.wkb_data_);
+            //size_ = std::move(other.size_);
+            geometry_ = std::move(other.geometry_);
+        }
+        return *this;
+    }
+
+    // operator std::string() const {
+    //     //tmp string created by copy ctr
+    //     return std::string(reinterpret_cast<const char*>(wkb_data_.get()),
+    //                        size_);
+    // }
+
+    // operator std::string_view() const {
+    //     return std::string_view(reinterpret_cast<const char*>(wkb_data_.get()),
+    //                             size_);
+    // }
+
+    ~Geometry() {
```

Comment:
The destructor uses OGRGeometryFactory::destroyGeometry(geometry_.release()) to free the underlying resource. While this works, a more professional approach is to use a custom deleter for the std::unique_ptr. This makes the ownership and cleanup logic more explicit and maintainable.

```
struct OGRGeometryDeleter {
    void operator()(OGRGeometry* ptr) {
        if (ptr) {
            OGRGeometryFactory::destroyGeometry(ptr);
        }
    }
};
```

Then, you would declare your geometry_ member with this custom deleter:

```std::unique_ptr<OGRGeometry, OGRGeometryDeleter> geometry_;```